iceoryx_utils/test/moduletests/test_semaphore_module.cpp

```diff
@@ -226,4 +230,23 @@ TEST_F(Semaphore_test, MoveCTorUnnamed)
     EXPECT_THAT(semaphore->post(), Eq(false));
 }
 
-#endif // not defined QNX
+TEST_F(Semaphore_test, TimedWaitWithTimeout)
+{
+    auto semaphore = iox::posix::Semaphore::create(0);
+    std::atomic<bool> timedWaitFinish{false};
+
+    std::thread t([&] {
+        struct timespec ts;
+        clock_gettime(CLOCK_REALTIME, &ts);
+        ts.tv_nsec += 2000000; // wait 2 ms
+        semaphore->timedWait(&ts, false);
+        timedWaitFinish.store(true);
+    });
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
You are right, but I really have to test the timed_wait that it actual waits! I could move it into an integration test but your argument would still be valid then. I could also remove this test, but then the sem_timedwait implementation would not be tested, but it is important to test it since in Windows we have a complex conversion into UTC.
@elBoberido what is your opinion?